File: nbrk_currency_rates/src/main/java/com/nbrk/rates/WidgetService.java
package com.nbrk.rates;

import android.content.Intent;
import android.widget.RemoteViewsService;

/**
 * Created by rpagyc on 30.10.2015.
 */
public class WidgetService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new ListProvider(this.getApplicationContext(), intent);
    }
}

File: nbrk_currency_rates/src/main/kotlin/com/nbrk/rates/widget/WidgetService.kt
package com.nbrk.rates.widget

import android.content.Intent
import android.widget.RemoteViewsService

/**
 * Created by rpagyc on 30.10.2015.
 */
class WidgetService : RemoteViewsService() {
  override fun onGetViewFactory(intent: Intent): RemoteViewsService.RemoteViewsFactory {
    return WidgetFactory(this.applicationContext, intent)
  }
}
